src/components/hint/__tests__/index.spec.tsx

```diff
@@ -1,69 +1,128 @@
+      ref={ref}
+    >
+      <Button round $backgroundDefault/>
+    </Hint>
   );
 };
 
+//TODO: Add test for onPress functionality
 describe('Hint Screen component test', () => {
-  afterEach(() => {
-    HintDriver.clear();
-  });
+  describe('Test Hint style:', () => {
+    it('Test Hint component background color', async () => {
+      const expectedColor = Colors.$backgroundPrimaryHeavy;
+      const renderTree = render(<HintTestComponent showHint/>);
+      const driver = HintDriver({renderTree, testID: TEST_ID});
+      expect(await driver.getElement()).toBeTruthy();
 
-  it('Test Hint component background color', async () => {
-    const expectedColor = Colors.$backgroundPrimaryHeavy;
-    const component = <HintTestComponent showHint/>;
+      let contentColor = await driver.getContentStyle().backgroundColor;
+      expect(contentColor).toBe(expectedColor);
 
-    const driver = new HintDriver({component, testID: 'Hint'});
-    expect(await driver.getElement()).toBeTruthy();
+      const whiteHintRenderTree = render(<HintTestComponent showHint color={Colors.white}/>);
+      const whiteHintDriver = HintDriver({renderTree: whiteHintRenderTree, testID: TEST_ID});
 
-    const hintContent = await driver.getHintContent();
-    let color = findStyle('backgroundColor', hintContent);
-
-    expect(color).toBe(expectedColor);
-    expect(hintContent).toBeTruthy();
+      contentColor = await whiteHintDriver.getContentStyle().backgroundColor;
+      expect(contentColor).toBe(Colors.white);
+    });
+  });
 
-    const WhiteHint = <HintTestComponent showHint color={Colors.white}/>;
-    const WhiteHintDriver = new HintDriver({component: WhiteHint, testID: 'Hint'});
+  describe('Test Hint icon', () => {
+    it('Hint should include icon', async () => {
+      const renderTree = render(<HintTestComponent showHint useIcon/>);
+      const driver = HintDriver({renderTree, testID: TEST_ID});
+      expect(driver.getIcon().exists()).toBeTruthy();
+    });
 
-    const WhiteHintContent = await WhiteHintDriver.getHintContent();
-    color = findStyle('backgroundColor', WhiteHintContent);
-    expect(color).toBe(Colors.white);
+    it('Hint shouldn\'t include icon', async () => {
+      const renderTree = render(<HintTestComponent showHint/>);
+      const driver = HintDriver({renderTree, testID: TEST_ID});
+      expect(driver.getIcon().exists()).toBeFalsy();
+    });
   });
 
-  it('Test Hint modal is not visible when showHint is false', async () => {
-    const component = <HintTestComponent showHint={false}/>;
-    const driver = new HintDriver({component, testID: 'Hint'});
-    expect(await driver.getHintModal()).toBeFalsy();
+  describe('Test Hint modal visibility:', () => {
+    it('Test Hint modal is not visible when showHint is false', async () => {
+      const renderTree = render(<HintTestComponent showHint={false}/>);
+      const driver = HintDriver({renderTree, testID: TEST_ID});
+      expect(await driver.getModal().exists()).toBeFalsy();
```

Comment:
A note, `exists` is a "more difficult" test than `visible`. you can exist but not be visible (if the user sends `useModal=false` for example the `Modal` will not exist).
It seems to me like if RN's `Modal` will change their implementation this can fail; while testing for `isVisible` should still work.